galsim/catalog.py

```diff
@@ -312,3 +317,334 @@ def itervalues(self):
+    # a comparison of the median absolute deviations to see which is better.
+    use_bulgefit = 1
+    if bstat<1 or bstat>4 or dvc_btt<0.1 or dvc_btt>0.9 or np.isnan(dvc_btt) or params[9]<=0 or \
+            params[1]<=0 or params[11]<0.051 or params[3]<0.051 or smad<bmad:
+        use_bulgefit = 0
+        # Then check if sersicfit is viable; if not, this object is a total failure:
+        if sstat<1 or sstat>4 or sparams[1]<=0 or sparams[0]<=0:
+            raise RuntimeError("Cannot make parametric model for this object")
+
+    # Now, if we're supposed to use the 2-component fits, get all the parameters.
+    if use_bulgefit:
+        # Bulge parameters.
+        bulge_q = params[11]
+        bulge_beta = params[15]*galsim.radians
+        bulge_hlr = 0.03*np.sqrt(bulge_q)*params[9]
+        bulge_flux = 2.0*np.pi*3.607*(bulge_hlr**2)*params[8]/0.03**2
+        # Disk parameters.
+        disk_q = params[3]
+        disk_beta = params[7]*galsim.radians
+        disk_hlr = 0.03*np.sqrt(disk_q)*params[1]
+        disk_flux = 2.0*np.pi*1.901*(disk_hlr**2)*params[0]/0.03**2
+        bfrac = bulge_flux/(bulge_flux+disk_flux)
+        # Make sure the bulge-to-total flux ratio is not nonsense.
+        if bfrac < 0 or bfrac > 1 or np.isnan(bfrac):
+            raise RuntimeError("Cannot make parametric model for this object")
+
+        # Then make the object.
+        if chromatic:
+            bulge = galsim.DeVaucouleurs(flux=1., half_light_radius=record['size_factor']*bulge_hlr) \
+                * sed_bulge.atRedshift(record['zphot']).withMagnitude(
+                record['mag_auto']-2.5*math.log10(bfrac*record['flux_factor']), _COSMOS_bandpass)
+            disk = galsim.Exponential(flux=1., half_light_radius=record['size_factor']*disk_hlr) \
+                * sed_disk.atRedshift(record['zphot']).withMagnitude(
+                record['mag_auto']-2.5*math.log10((1.-bfrac)*record['flux_factor']), _COSMOS_bandpass)
+        else:
+            bulge = galsim.DeVaucouleurs(flux = record['flux_factor']*bulge_flux,
+                                         half_light_radius = record['size_factor']*bulge_hlr)
+            disk = galsim.Exponential(flux = record['flux_factor']*disk_flux,
+                                      half_light_radius = record['size_factor']*disk_hlr)
+        # Apply shears for intrinsic shape.
+        if bulge_q < 1.:
+            bulge = bulge.shear(q=bulge_q, beta=bulge_beta)
+        if disk_q < 1.:
+            disk = disk.shear(q=disk_q, beta=disk_beta)
+        return bulge+disk
+    else:
+        (fit_gal_flux, fit_gal_hlr, fit_gal_n, fit_gal_q, _, _, _, fit_gal_beta) = sparams
+
+        gal_n = fit_gal_n
+        # Fudge this if it is at the edge of the allowed n values.  Since GalSim (as of #325 and
+        # #449) allow Sersic n in the range 0.3<=n<=6, the only problem is that the fits
+        # occasionally go as low as n=0.2.
```

Comment:
I thought there were some with n>6 as well.  No?  Even if not, it might be worth adding a guard for that as well to use n=6 if an input has a larger n, just in case a future version of the catalog has some of these.